### PR TITLE
test/downgrade: Convert the single node stage to a downgrade test

### PR DESCRIPTION
**Pull request by sayf-eddine-scality**
Fixes: #1697

**Component**:
upgrade, downgrade, tests

**Context**: 
Upgrade/downgrade are not tested by eve

**Summary**:
We run downgrade tests to N-1 version in the CI

**Acceptance criteria**: 
The upgrade stage should be successful 

---

Closes: #1697

### CODE REVIEW

File: eve/main.yml [hunk 1]
```diff
@@ -216,7 +224,6 @@ stages:
           command: >
             sudo mount -o loop metalk8s.iso
             /srv/scality/metalk8s-%(prop:metalk8s_version)s
-          haltOnFailure: true
```
Comment (Ebaneck) [inline eve/main.yml]: I think we should always keep `haltOnFailure: true`.. no??

File: eve/main.yml [hunk 2]
```diff
@@ -277,7 +284,34 @@ stages:
...
+            shasumurl="/pre/SHA256SUM-%(prop:metalk8s_prev_version)s";
+            curl -s -XGET -o /tmp/SHA256SUM
+            "%(prop:artifacts_private_url)s$shasumurl"
+          haltOnFailure: true
+      - ShellCommand:
+          name: Check N-1 image with checksum
+          command: cd /tmp && sha256sum -c SHA256SUM
+          haltOnFailure: true
           alwaysRun: true
+      - ShellCommand:
```
Comment (Ebaneck) [inline eve/main.yml]: Ditto, we should add `haltOnFailure: true`

File: eve/main.yml [hunk 3]
```diff
@@ -277,7 +284,34 @@ stages:
...
+          command: >
+            shasumurl="/pre/SHA256SUM-%(prop:metalk8s_prev_version)s";
+            curl -s -XGET -o /tmp/SHA256SUM
+            "%(prop:artifacts_private_url)s$shasumurl"
+          haltOnFailure: true
+      - ShellCommand:
+          name: Check N-1 image with checksum
+          command: cd /tmp && sha256sum -c SHA256SUM
+          haltOnFailure: true
           alwaysRun: true
```
Comment (TeddyAndrieux) [inline eve/main.yml]: I think you don't want to alwaysRun this one

File: eve/main.yml (hunk 1 above)
Comment (TeddyAndrieux) [reply eve/main.yml]: Agree

File: eve/main.yml [hunk 4]
```diff
@@ -277,7 +284,34 @@ stages:
...
+      - ShellCommand:
+          name: Run downgrade
+          command: >
+            sudo bash
+            /srv/scality/metalk8s-%(prop:metalk8s_version)s/iso-manager.sh
+            --archive /tmp/metalk8s.iso &&
+            sudo bash
+            /srv/scality/metalk8s-%(prop:metalk8s_version)s/downgrade.sh
+            --destination-version %(prop:metalk8s_prev_version)s
```
Comment (TeddyAndrieux) [inline eve/main.yml]: `haltOnFailure: true` for this one make sense IMO

**Review by TeddyAndrieux — DISMISSED** [review]

File: eve/main.yml [hunk 5]
```diff
@@ -277,7 +284,34 @@ stages:
...
+          name: Retrieve ISO N-1 image checksum
+          command: >
+            shasumurl="/pre/SHA256SUM-%(prop:metalk8s_prev_version)s";
+            curl -s -XGET -o /tmp/SHA256SUM
+            "%(prop:artifacts_private_url)s$shasumurl"
+          haltOnFailure: true
+      - ShellCommand:
+          name: Check N-1 image with checksum
+          command: cd /tmp && sha256sum -c SHA256SUM
+          haltOnFailure: true
```
Comment (TeddyAndrieux) [inline eve/main.yml]: IMO you should retrieve and check both ISO at the very beginning (and also you should use the same yaml tag for upgrade and downgrade)

File: eve/main.yml [hunk 6]
```diff
@@ -277,7 +284,34 @@ stages:
       - Upload:
           name: upload artifacts
           source: sosreport
```
Comment (TeddyAndrieux) [inline eve/main.yml]: Like in Upgrade please move `upload artifacts` and sosreport to the end of the step (`alwaysRun:true` so we will always have the report

Comment (sayf-eddine-scality) [reply eve/main.yml]: I wanted to have 2 sos report collections if someday we want to see the difference between the two states

Comment (TeddyAndrieux) [reply eve/main.yml]: Then please rename the file like `sosrepot-blabla-post-install.tar.xz` `sosreport-blabla-post-upgrade.tar.xz` `sosreport-blabla-post-downgrade`

File: eve/main.yml [hunk 7]
```diff
@@ -277,7 +313,6 @@ stages:
...
-          alwaysRun: true
```
Comment (TeddyAndrieux) [inline eve/main.yml]: Keep always run I think

File: eve/main.yml [hunk 8]
```diff
@@ -177,22 +177,30 @@ stages:
...
-  single-node:
+  single-node-downgrade-test:
     worker:
       type: openstack
       image: CentOS-7-x86_64-GenericCloud-1809.qcow2
       flavor: m1.large
       path: eve/workers/openstack-single-node
     steps:
       - Git: *git_pull
+      - SetPropertyFromCommand:
```
Comment (TeddyAndrieux) [inline eve/main.yml]: Please use yaml anchor (`&set_prev_vers` `*set_prev_vers`) for all common step for downgrade/upgrade to remove copy paste. At least :
- `Set previous version`
- `Retrieve ISO version N-1 image`
- `Retrieve ISO N-1 image checksum`
- `Check N-1 image with checksum`

Comment (Ebaneck) [reply eve/main.yml]: @TeddyAndrieux lets merge one first and then we can add and re-use `anchors` in the second PR. This is the reason why the second PR on `Upgrade` is in draft mode till now

Comment (TeddyAndrieux) [reply eve/main.yml]: Ok yes you can add/use them in a single PR, as you want

Comment (TeddyAndrieux) [reply eve/main.yml]: @sayf-eddine-scality please do this, as the upgrade merged now :)

**Review by Ebaneck — APPROVED** [review]

**Pull request merged**